pkg/alertmanager/amcfg_test.go

```diff
@@ -2939,6 +2942,57 @@ func TestSanitizeConfig(t *testing.T) {
 			},
 			expectErr: true,
 		},
+		{
+			name:           "message_thread_id field for Telegram config",
+			againstVersion: versionTelegramMessageThreadIDAllowed,
+			in: &alertmanagerConfig{
+				Receivers: []*receiver{
+					{
+						Name: "telegram",
+						TelegramConfigs: []*telegramConfig{
+							{
+								ChatID:          12345,
+								MessageThreadID: 123,
+								BotTokenFile:    "/test",
+							},
+						},
+					},
+				},
+			},
+			expect: alertmanagerConfig{
+				Receivers: []*receiver{
+					{
+						Name: "telegram",
+						TelegramConfigs: []*telegramConfig{
+							{
+								ChatID:          12345,
+								MessageThreadID: 123,
+								BotTokenFile:    "/test",
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name:           "message_thread_id is dropped for unsupported versions",
+			againstVersion: versionTelegramMessageThreadIDNotAllowed,
+			in: &alertmanagerConfig{
+				Receivers: []*receiver{
+					{
+						Name: "telegram",
+						TelegramConfigs: []*telegramConfig{
+							{
+								ChatID:          12345,
+								MessageThreadID: 123,
+								BotTokenFile:    "/test",
+							},
+						},
+					},
+				},
+			},
+			expectErr: true,
```

Comment:
the returned error isn't the one expected (it says `missing mandatory field botToken or botTokenFile`). You need to set `botToken` instead.

Reply:
@simonpasquier Can you just support me here? The changes above broke my tests..?